cognite/neat/_graph/transformers/_base.py

```diff
@@ -12,3 +16,121 @@ class BaseTransformer(ABC):
     @abstractmethod
     def transform(self, graph: Graph) -> None:
         raise NotImplementedError()
+
+
+class BaseTransformerStandardised(ABC):
+    """Standardised base transformer to use in case a transformer is adding or removing triples from a graph. If you
+    are doing more specialised operations, please overwrite the .transform() method.
+    """
+
+    _use_only_once: bool = False
+    _need_changes: ClassVar[frozenset[str]] = frozenset()
+    _use_iterate_bar_threshold: int = 500
+
+    @abstractmethod
+    def description(self) -> str:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def operation(self) -> Literal["add", "remove"]:
+        raise NotImplementedError()
+
+    def _target_properties_count_query(self) -> str | None:  # noqa: B027
+        """
+        Overwrite to fetch all affected properties in the graph as a result of the transformation.
+        Returns:
+            A query string.
+        """
+        ...
+
+    def _target_edges_count_query(self) -> str | None:  # noqa: B027
+        """
+        Overwrite to fetch all affected edges (objects) in the graph
+        as a result of the transformation.
+        Returns:
+            A query string.
+        """
+        ...
+
+    def _skip_query(self) -> str | None:  # noqa: B027
+        """
+        Overwrite to fetch all affected triples (subjects, objects and predicates) in
+        the graph as a result of the transformation.
+        Returns:
+            A query string.
+        """
+        ...
+
+    @abstractmethod
+    def _iterate_query(self) -> str:
+        """
+        The query to use for extracting target triples from the graph and performing the transformation.
+        Returns:
+            A query string.
+        """
+        raise NotImplementedError()
+
+    def transform(self, graph: Graph) -> GraphTransformationResult:
+        added_entities = []
+        removed_entities = []
+        skipped_entities = []
+        affected_nodes_count = 0
+        issues: IssueList = IssueList([])
+
+        if self._skip_query():
+            skipped_entities = graph.query(self._skip_query())
+            skipped_entities = [str(triple) for triple in skipped_entities]
+
+        targets = list(graph.query(self._iterate_query()))
```

Comment:
This fill materialize every result into memory. Move to the iteration loop.

To count target, use the target_properties_query.